src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1479,6 +1486,124 @@ class Simplex_tree {
     }
   }
 
+  /** \brief Retrieve good values for extended persistence, and separate the 
+   * diagrams into the ordinary, relative, extended+ and extended- subdiagrams. 
+   * \pre This function should be called only if `extend_filtration()` has been called first!
+   * \post The coordinates of the persistence diagram points might be a little different than the
+   * original filtration values due to the internal transformation (scaling to [-2,-1]) that is 
+   * performed on these values during the computation of extended persistence.
+   * @param[in] dgm Persistence diagram obtained after calling `extend_filtration()`, 
+   * `initialize_filtration()`, and `Gudhi::persistent_cohomology::Persistent_cohomology< FilteredComplex, CoefficientField >::compute_persistent_cohomology()`.
+   * @param[in] efd Structure containing the minimum and maximum values of the original filtration
+   * @return A vector of four persistence diagrams. The first one is Ordinary, the 
+   * second one is Relative, the third one is Extended+ and the fourth one is Extended-.
+   * See section 2.2 in https://link.springer.com/article/10.1007/s10208-017-9370-z for a description of these subtypes.
+   */
+  std::vector<std::vector<std::pair<int, std::pair<Filtration_value, Filtration_value>>>> extended_persistence_subdiagrams(const std::vector<std::pair<int, std::pair<Filtration_value, Filtration_value>>>& dgm, const Extended_filtration_data& efd){
+    std::vector<std::vector<std::pair<int, std::pair<Filtration_value, Filtration_value>>>> new_dgm(4); 
+    Filtration_value x, y;
+    Filtration_value minval = efd.minval;
+    Filtration_value maxval = efd.maxval;
+    for(unsigned int i = 0; i < dgm.size(); i++){ 
+      int h = dgm[i].first; 
```

Comment:
I wonder if it is more common to keep the dimension, or to correct it, say use h-1 as dimension for features in the relative pass?

Reply:
I would rather keep it like this, and maybe refer to something, for instance the symmetry theorem in https://link.springer.com/article/10.1007/s10208-008-9027-z ?

Reply:
I just wanted to check if the coning did something weird to the dimension. In a graph, a downward branch will appear as dim 0 while an upward branch appears as dim 1, which matches what relative homology says. So this looks good as is.